File: ARCADE/Intro/Level 5_4: avoidObstacles.cs
    using System;
     
    public class Test
    {
    	public static int avoidObstacles(int[] inputArray){
            for (int i = 1; ; i++){
                int count = 0;
                for (int j = 0; j < inputArray.Length; j++){
                    if (inputArray[j] % i == 0){
                        break;
                    }
                    count++;
                    if (count >= inputArray.Length){
                        return i;
                    }
                }
            }
        }
    	public static void Main()
    	{
    		int [] inputArray = new int [] {5, 3, 6, 7, 9};
    		Console.WriteLine(avoidObstacles(inputArray));
    	}
    }

File: ARCADE/Intro/Level 5_4: avoidObstacles.pas
program avoidObstacles;
 
Type
	ArrayInteger = Array of integer;
 
function avoidObstacles(inputArray: ArrayInteger): integer;
var i, j, rs, count : integer;
begin
    rs := 0;
    for i:= 1 to 40 do
    begin
        count := 0;
        for j := 0 to length(inputArray) - 1 do
        begin
             if(inputArray[j] mod i = 0) then break;
             count := count + 1;
             if(count = length(inputArray)) then
             begin
                 rs := i;
             end;
        end;
        if(rs <> 0) then break;
    end;
    avoidObstacles := rs;
end;
 
var inputArray : ArrayInteger;
begin
	inputArray := ArrayInteger.Create(5, 3, 6, 7, 9);
	writeln(avoidObstacles(inputArray));
end.
